### DISCUSSION TITLE
Frigate LXC with nvidia driver support

### DISCUSSION THREAD

**Discussion by xijio** [community-scripts/ProxmoxVE#966] [General]
I briefly had the @remz1337 fork of the frigate script working with proxmox using NVIDIA-Linux-x86_64-550.135.run on the host using "bash -c "$(wget -qLO - https://github.com/remz1337/Proxmox/raw/remz/ct/frigate.sh)"".  But then I was having trouble with the nvidia driver in another lxc and removed the 550.135 drivers and installed the proxmox nvidia-driver from apt (535.183.01). This broke the @remz1337 installed Frigate due to driver mismatch.  I've been trying for a full day now to get either the lxc to match the driver of the host, or move the host back to 550.135 drivers and I'm faceplanting hard.

Passing GPUs to LXCs just seems really complicated.  The @remz1337 fork worked for a while, but broke when I made a change and now when I try to run again to redeploy it errors out.  I was going to post in that repo, but it has been archived as of the 17th.  This brings me to my question:

Is there a possiblity of getting a frigate lxc script with nvidia driver support to be a part of the repo?  If not, what is the recommended way to enable nvidia graphic support on the current script?

Thank you in advance!

**Comment by remz1337** [comment]
hey there, it's me. The original repo (tteck) was archived and so is now my fork of it too. You can still find my frigate script with gpu support on my new fork of the community-scripts repo (https://github.com/remz1337/ProxmoxVE/).

Indeed, for gpu passthrough to work in any LXC, drivers need to match exactly the host.

I have a script to install drivers across host and LXC:
`bash -c "$(wget -qLO - https://github.com/remz1337/ProxmoxVE/raw/remz/misc/nvidia-drivers-host.sh)"`

If you want an older version, download the file to your host and change this [line 68](https://github.com/remz1337/ProxmoxVE/blob/23327f03629d7fc84130784ac83ea14c7110edb9/misc/nvidia-drivers-host.sh#L68) for the version you want to install (make sure it is listed here: https://download.nvidia.com/XFree86/Linux-x86_64/)

**Comment by xijio** [comment]
Hi Remz, I totally missed this over Christmas!  Thanks a ton for replying to me, I really appreciate it!  I think your nvidia driver script is really cool and might be considered as an install option in a bunch of different LXCs in the future.  As we move more and more toward AI GPUS are getting more and more popular in selfhosted applications and I think it'd be great to be able to easily manage the drivers on the host and in each lxc with community scripts.  For now though I'll definitely take a look at the nvidia script.  Thanks a ton!

**Comment by petertoth-dev** [comment]
@remz1337 For me your script just doesn't work. I ran on host, set the latest driver, selected the frigate container, installed the drivers there. Saw no errors, but when I ssh in nvidia-smi says: `NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.`

```
# lspci | grep VGA
03:00.0 VGA compatible controller: NVIDIA Corporation TU106 [GeForce RTX 2060 SUPER] (rev a1)
```

**Reply by remz1337** [reply]
to troubleshoot this, try to do things manually first. Download and install on host, make sure nvidia-smi works, then passthrough the gpu to one LXC, install the drivers in the LXC and see if nvidia-smi works in the LXC.

not sure if host reboot is needed, but might be worth trying after the driver install on the host

**Reply by petertoth-dev** [reply]
I ended up doing everything manually and that way it worked, but the script didn't make the magic for me. Thank you for the script anyways, it's an incredible job.
